### PR TITLE
Fixes #313, #283, #322, #127, #334 - Add button to remove conference from recents list

### LINKED ISSUE #313: Deleting or bookmarking old session
When we launch jitsi-meet-electron, we can see our last "conference room". 
Is there a way to remove some or all of them?  Is there a way to create favorite ones?

### PR DESCRIPTION
**Pull request by freddytuxworth**
This PR makes it possible to remove conferences from the recents list, for convenience/privacy.
There are a bunch of issues about it: #313, #283, #322, #127, #334.

### CODE REVIEW

**Comment by awlx** [comment]
Nice! This will make many people very happy! :)

**Comment by krombel** [comment]
Isn't it more wise to implement it in https://github.com/jitsi/jitsi-meet? Or just "backport" it there so it is available for especially the web people (which are as well complaining about this missing feature).
[This](https://github.com/jitsi/jitsi-meet/blob/stable/jitsi-meet_4627/react/features/recent-list/components/RecentList.web.js) looks at the first look at a good starting point.
Any plans/insights on that?

**Comment by freddytuxworth** [comment]
> Isn't it more wise to implement it in https://github.com/jitsi/jitsi-meet? Or just "backport" it there so it is available for especially the web people (which are as well complaining about this missing feature).
> [This](https://github.com/jitsi/jitsi-meet/blob/stable/jitsi-meet_4627/react/features/recent-list/components/RecentList.web.js) looks at the first look at a good starting point.
> Any plans/insights on that?

As far as I can see those are two completely separate recent lists, so it would need to be implemented separately in both places. Happy to do that though, I'll give it a go :)

**Comment by awlx** [comment]
They recent list in the electron-client is kept separately I guess, because it also keeps track of different servers which die web UI can't do.

**Comment by awlx** [comment]
The feature is working really well :). 

I merged it our fork for testing! It makes many users very happy!
Screenshot 2020-06-13 at 11 14 35

File: app/features/recent-list/reducer.js [hunk 1]
```diff
@@ -65,6 +72,21 @@ function _insertConference(
...
+ * @param {Array<RecentListItem>} recentList - Previous recent list array.
+ * @param {RecentListItem} toRemove - Conference to be removed.
+ * @returns {Array<RecentListItem>} - Updated recent list array.
+ */
+function _removeConference(
+        recentList: Array<RecentListItem>,
+        toRemove: RecentListItem
+): Array<RecentListItem> {
+    return recentList.filter(
+        (conference: RecentListItem) => conference.room !== toRemove.room);
```
Comment (saghul) [inline app/features/recent-list/reducer.js:87]: You also need to check the serverURL field, or you'd end up removing all conferences with the same name, no matter the server.

But actually, why not remove all elements which are not `toRemove` ?

File: app/features/recent-list/components/RecentList.js [hunk 2]
```diff
@@ -5,14 +5,16 @@ import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import type { Dispatch } from 'redux';
 import { push } from 'react-router-redux';
-
```
Comment (saghul) [inline app/features/recent-list/components/RecentList.js:8]: please leave this newline

File: app/features/recent-list/reducer.js (hunk 1 above)
Comment (freddytuxworth) [reply app/features/recent-list/reducer.js]: Good point, not sure why I added the `.room`s

**Comment by freddytuxworth** [comment]
Should be fixed :+1:

**Comment by benczur45** [comment]
Hello there!

Is this available on Browser version too? 

Thanks in advance.

**Comment by saghul** [comment]
No, that is a separate implementation.

**Comment by luistriana28** [comment]
@saghul 

Is there gonna be an update for web browser?

and for dockerized one too?

**Comment by saghul** [comment]
Removing items for the recent list on the web interface just landed in master today. It will be part of the next stable release.

**Pull request merged**